list.go

```diff
@@ -101,6 +101,19 @@ func NewList() *List {
 	}
 }
 
+// SelectItem calls the given item's select function (if defined).
+func (l *List) SelectItem(index int) {
+	if index >= 0 && index < len(l.items) {
+		item := l.items[index]
+		if item.Selected != nil {
+			item.Selected()
+		}
+		if l.selected != nil {
+			l.selected(index, item.MainText, item.SecondaryText, item.Shortcut)
+		}
+	}
+}
+
```

Comment:
```suggestion
func (l *List) SelectItem(index int) {
	if index < 0 || index >= len(l.items) {
		return
	}
	
	item := l.items[index]
	if item.Selected != nil {
		item.Selected()
	}
	if l.selected != nil {
		l.selected(index, item.MainText, item.SecondaryText, item.Shortcut)
	}
}

```


But maybe this could simply work

```suggestion
func (l *List) SelectItem(index int) {
	item, found := l.items[index]
	if !found {
		return
	}
	if item.Selected != nil {
		item.Selected()
	}
	if l.selected != nil {
		l.selected(index, item.MainText, item.SecondaryText, item.Shortcut)
	}
}

```

Reply:
I generally agree with your stylistic choices on the code, but I was trying to keep the diff as minimal as possible (ie. just moving what the original author already wrote and not re-write their code and introduce the possibility of mistakes introducing new and unexpected behaviors). Happy to change it if the maintainer wants though, I do prefer yours and this is simple enough.